federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -180,11 +178,11 @@ func (cc *ClusterController) UpdateClusterStatus() error {
 		return err
 	}
 
-	cc.mu.Lock()
+	cc.Lock()
+	defer cc.Unlock()
```

Comment:
i thought about this in the original change, but not too sure about it since it can potentially holds the lock for a long time when it tries to update the clusters' status ? But perhaps its not too bad since this reconciliation loop is not run very frequently i think ?